async-aggregators/lib/hidden_in_ascii/hidden_in_ascii.py

```diff
@@ -0,0 +1,183 @@
+import random
+from pathlib import Path
+from typing import List, Tuple
+
+import numpy as np
+from PIL import Image, ImageDraw
+
+
+def prepare_input(img_path: Path) -> Tuple[Image.Image, List[int]]:
+    """
+    Prepare the input image by center cropping into a square
+
+    :param img_path: input image file path
+    :return: Tuple of cropped input Image and the coordinates of the cropped area
+    """
+    img = Image.open(img_path)
+
+    # Calculate the square size
+    width, height = img.size
+    size = min(width, height)
+
+    # Calculate the coordinates for cropping
+    left = (width - size) // 2
+    top = (height - size) // 2
+    right = (width + size) // 2
+    bottom = (height + size) // 2
+
+    # Crop the image to a square
+    return img.crop((left, top, right, bottom)), [left, top, right, bottom]
+
+
+def avg_greyscale(img: Image.Image) -> float:
+    """
+    Calculates average greyscale or luminance value of an input image
+
+    :param img: input image
+    :return: average luminance
+    """
+    np_img = np.array(img)
+    w, h = np_img.shape
+    return np.average(np_img.reshape(w * h))
+
+
+def img_to_ascii(img: Image.Image, dens: int) -> List[str]:
```

Comment:
Since 3.9, you can use the built-in types here (note the capitalization of the typehint).
```suggestion
def img_to_ascii(img: Image.Image, dens: int) -> list[str]:
```